### PR TITLE
Add a Botan BSI CI

### PR DESCRIPTION
**Pull request by lieser**
This adds a workflow to run a BSI specific CI for Botan.

- Add CI for PDF Handbook and Doxygen API documentation, and provide it as an artifact
- Add CI for coverage report, and provide it as an artifact
- Add a CI to test Botan with BSI specific config

Most is adapted from the CI in [Rohde-Schwarz/botan.](https://github.com/Rohde-Schwarz/botan)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
+name: Release
+
+on:
+  - workflow_dispatch
+  # TODO: Temporary, should be removed before merged to main
```
Comment (reneme) [inline .github/workflows/release.yml:5]: Remember TODO

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        with:
+          target: ${{ matrix.target }}
+          cache-key: bsi-${{ matrix.host_os }}-${{ matrix.target }}
+
+      - name: Build and Test Botan
+        # TODO: Disabled tests is temporary fix for https://github.com/randombit/botan/pull/3632
+        run: >-
+          python3 ./source/.github/scripts/ci_build.py
+          --root-dir=${{ github.workspace }}/botan --build-dir=${{ github.workspace }}/build
+          --disabled-tests=tls_extensions_key_share_client_hello,xmss_keygen_reference
```
Comment (reneme) [inline .github/workflows/release.yml:60]: `xmss_keygen_reference` could be fixed by enabling `shake`.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+          target: ${{ matrix.target }}
+          cache-key: bsi-${{ matrix.platform.host_os }}-${{ matrix.platform.compiler }}-${{ matrix.target }}
+
+      - name: Build and Test Botan
+        # TODO: Disabled tests is temporary fix for https://github.com/randombit/botan/pull/3632
+        run: >-
+          python3 ./source/.github/scripts/ci_build.py
+          --root-dir=${{ github.workspace }}/botan --build-dir=${{ github.workspace }}/build
+          --cc='${{ matrix.platform.compiler }}' --make-tool='${{ matrix.platform.make_tool }}'
+          --disabled-tests=tls_extensions_key_share_client_hello,xmss_keygen_reference
```
Comment (reneme) [inline .github/workflows/release.yml:112]: As above, `xmss_keygen_reference` just lacks `shake`. We don't have to disable this.

**Review by reneme — COMMENTED** [review]
No thorough review, just some remarks.

Question: Is the `.github/actions/setup-build-agent` a verbatim copy of the action in `randombit/botan`? If yes, we may be able to just invoke it in our `release.yml`. But I guess you would have done that if it were. :)

File: .github/workflows/release.yml (hunk 2 above)
Comment (lieser) [reply .github/workflows/release.yml]: `shake` is now enabled indirectly by enabling both kyber and sphincsplus.

File: .github/scripts/ci_build.py [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+        flags += ['--no-store-vc-rev', '--compiler-cache=%s' % (ccache)]
+
+    flags += ['--werror-mode']
+
+    enable_modules = []
+    enable_modules += ['tls12','tls13','tls_cbc']
+    enable_modules += ['pkcs11']
+    enable_modules += ['xts']
+    enable_modules += ['kyber','kyber_90s']
+    enable_modules += ['sphincsplus_sha2','sphincsplus_shake']
```
Comment (reneme) [inline .github/scripts/ci_build.py:105]: No `dilithium` and `dilithium_aes`?

**Review by reneme — CHANGES_REQUESTED** [review]

File: .github/scripts/ci_build.py (hunk 4 above)
Comment (lieser) [reply .github/scripts/ci_build.py]: added

**Review by reneme — APPROVED** [review]
Build failed due to the same `jom` issue as currently in the Botan CI. 😠 Otherwise: LGTM.

File: .github/workflows/release.yml (hunk 1 above)
Comment (lieser) [reply .github/workflows/release.yml]: Removed

**Pull request merged**